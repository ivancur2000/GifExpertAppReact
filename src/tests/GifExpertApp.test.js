import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";


describe('Prueba del componente GifExpertApp',()=>{

    test('debe mostrar el snapShot', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe mostrar la lista de categorias', () => {
        
        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);

    });
    

});