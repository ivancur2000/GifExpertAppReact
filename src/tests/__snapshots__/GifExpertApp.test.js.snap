// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del componente GifExpertApp debe mostrar el snapShot 1`] = `
<Fragment>
  <h2>
    Gif Expert
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="full metal"
      key="full metal"
    />
  </ol>
</Fragment>
`;

exports[`Prueba del componente GifExpertApp debe mostrar la lista de categorias 1`] = `
<Fragment>
  <h2>
    Gif Expert
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
